Assignment-1/FetchAPI.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+require __DIR__ . '/requests.php';
+
+class FetchAPI {
+  private static $domain = 'https://www.innoraft.com/';
+  private $url;
+  public $title = [];
+  public $title_img = [];
+  public $icon_arrays = [];
+  public $details = [];
+
+  /**
+   * Constructor function for setting url.
+   *
+   * @param string $api_url
+   *   For setting the $url.
+   */
+  function __construct (string $api_url) {
+    $this->url = $api_url;
+  }
+
+  /**
+   *  Get data from API url and set tha data to class variables.
+   */
+  public function set_data() {
+    try {
+      // API response data.
+      $response_data = request($this->url);
+      if ($response_data == NULL) {
+        throw new Exception;
```

Comment:
When you are throwing an exception, you should pass an error message with it.
Then in the catch statment you can echo that exception.
But beware, you should not reveal any technical error messages to the end user.